### PR TITLE
[feat] Downloads UI updates

### PR DESCRIPTION
**Pull request by matejmolnar**
## Changes
- Update the Downloads screen UI to be similar to the Uploads UI
- On appear load active download tasks

https://github.com/strvcom/ios-networking/assets/17743607/075ddc20-1cef-4eb5-b169-55aa348b8ef4

### CODE REVIEW

**Review by cejanen — CHANGES_REQUESTED** [review]
Nice finding and proactive step. Two minor things worth to look at.

File: NetworkingSampleApp/NetworkingSampleApp/Scenes/Download/DownloadProgressView.swift [hunk 1]
```diff
@@ -11,65 +11,76 @@ struct DownloadProgressView: View {
...
-                    viewModel.cancel()
-                } label: {
-                    Text("Cancel")
+
+                HStack {
+                    ProgressView(value: viewModel.state.percentCompleted, total: 100)
+                        .progressViewStyle(.linear)
+                        .frame(width: 150)
+
+                    Text("\(String(format: "%.1f", viewModel.state.megaBytesCompleted))MB")
```
[the remark this answers is not in the document]
Comment (matejmolnar) [reply NetworkingSampleApp/NetworkingSampleApp/Scenes/Download/DownloadProgressView.swift]: Good idea, I didn't notice that 👍

File: NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import SwiftUI
+
+struct TaskButton: View {
+    enum Config {
+        case play, pause, cancel, retry
+
+        var imageName: String {
+            switch self {
+            case .play:
+                return "play"
```
Comment (cejanen) [inline NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift:17]: do we need return here?

Comment (cejanen) [reply NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift:17]: similar for all switches about Self?

**Review by cejanen — CHANGES_REQUESTED** [review]
One more comment, if it's possible to update

File: NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift (hunk 2 above)
Comment (matejmolnar) [reply NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift]: Done ✅

File: NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            switch self {
+            case .play, .pause, .retry:
+                .blue
+            case .cancel: 
+                .red
+            }
+        }
+    }
+
+    let config: Config
```
Comment (cejanen) [inline NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift:33]: I'm always challenging the public properties even though let. Does it make sense for any other structures to have those two visible?

**Review by cejanen — DISMISSED** [review]
I left comment which is more subjective so we're good to go.

🍝

File: NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift (hunk 3 above)
Comment (matejmolnar) [reply NetworkingSampleApp/NetworkingSampleApp/UIElements/TaskButton.swift]: True, I made them private.

**Review by cejanen — APPROVED** [review]
🦧

**Review by johnkodes — APPROVED** [review]

**Pull request merged**